planner/core/expression_rewriter.go

```diff
@@ -865,7 +865,13 @@ func (er *expressionRewriter) rewriteVariable(v *ast.VariableExpr) {
 			return
 		}
 	}
-	if v.IsGlobal {
+	sysVar := variable.SysVars[name]
+	if sysVar == nil {
+		er.err = errors.Trace(variable.UnknownSystemVar.GenWithStackByArgs(name))
```

Comment:
no need to call `errors.Trace()` since the error is generated with a call stack.